### PR TITLE
Intel_LLVM compiler macros

### PR DESCRIPTION
**Pull request by napetrov**
# Description
Update intel compiler macros checks with Intel_LLVM case for migration to ICX compiler

### CODE REVIEW

**Comment by napetrov** [comment]
Restarted run without sporadic errors. 
http://intel-ci.intel.com/ede39c7a-825a-f1fc-81ac-a4bf010d0e2e

With @KulikovNikita fix this seems to be working fine for bazel

File: cpp/daal/include/services/daal_defines.h [hunk 1]
```diff
@@ -50,7 +50,8 @@
 #endif
 
 /* Intel(R) oneDAL 64-bit integer types */
-#if (!defined(__INTEL_COMPILER)) & defined(_MSC_VER)
+
+#if (!defined(__INTEL_COMPILER) || !defined(__INTEL_LLVM_COMPILER)) && defined(_MSC_VER)
```
Comment (ahuber21) [inline cpp/daal/include/services/daal_defines.h:54]: Could you spell out the condition for me? This truth table looks unexpected.
image

Are we sure we're not checking for

```c
!(defined(__INTEL_COMPILER) || defined(__INTEL_LLVM_COMPILER))
```

File: cpp/daal/src/algorithms/service_kernel_math.h [hunk 2]
```diff
@@ -396,33 +396,7 @@ class MinkowskiDistances : public PairwiseDistances<FPType, cpu>
...
-            {
-                d += math.sPowx(math.sFabs(x[i] - y[i]), _p);
-            }
-
-            if (!_powered) return math.sPowx(d, 1.0 / _p);
-
-            return d;
-        }
-    }
+    inline FPType computeDistance(const FPType * x, const FPType * y, const size_t n) const;
```
Comment (ahuber21) [inline cpp/daal/src/algorithms/service_kernel_math.h:399]: Why are the `inline`s necessary. Just to fix comile errors?

Comment (napetrov) [reply cpp/daal/src/algorithms/service_kernel_math.h:399]: yes

**Review by ahuber21 — APPROVED** [review]

File: cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_predict_dense_default_batch_impl.i [hunk 3]
```diff
@@ -694,7 +694,7 @@ Status PredictClassificationTask<algorithmFPType, cpu>::predictOneRowByAllTrees(
     return safeStat.detach();
 }
 
-#if defined(__INTEL_COMPILER)
```
Comment (ahuber21) [inline cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_predict_dense_default_batch_impl.i:697]: The check for `__INTEL_COMPILER` is sometimes replaced with `__AVX512F && DAAL_INTEL_CPP_COMPILER` and sometimes only with `DAAL_INTEL_CPP_COMPILER`. Why?

**Comment by napetrov** [comment]
@ahuber21  - sorry, noticed that i've mess-up with rebase initially. now this pr have correct content . 
1. initial macros 
2. fix for compilation
3. your fix for not intel MSVS defines line

File: cpp/daal/src/algorithms/k_nearest_neighbors/kdtree_knn_classification_train_dense_default_impl.i [hunk 4]
```diff
@@ -39,7 +39,7 @@
 #include "src/algorithms/k_nearest_neighbors/kdtree_knn_impl.i"
 #include "src/algorithms/engines/engine_batch_impl.h"
 
-#if defined(__INTEL_COMPILER_BUILD_DATE)
```
Comment (ahuber21) [inline cpp/daal/src/algorithms/k_nearest_neighbors/kdtree_knn_classification_train_dense_default_impl.i:42]: How does `__INTEL_COMPILER_BUILD_DATE` map to `DAAL_INTEL_CPP_COMPILER`? Was it previously simply a synonym to `__INTEL_COMPILER`? Or is there more to pay attention to?

Comment (napetrov) [reply cpp/daal/src/algorithms/k_nearest_neighbors/kdtree_knn_classification_train_dense_default_impl.i:42]: yes.

File: cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_predict_dense_default_batch_impl.i (hunk 3 above)
Comment (napetrov) [reply cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_predict_dense_default_batch_impl.i:697]: as far i see there is behavior change between compilers and ICX is trying to compiler avx512 code even if it wasn't targeted for this code.  So AVX512 defines are put in place where we have direct AVX512 usage

**Pull request merged**